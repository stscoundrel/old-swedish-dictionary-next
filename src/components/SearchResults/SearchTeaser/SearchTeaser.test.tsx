import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import SearchTeaser from './index'

const result = {
  headword: 'afköra',
  partOfSpeech: 'vb',
  grammaticalAspect: 'v.',
  definitions: [
    ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
  ],
  alternativeForms: ['-bortdrifva'],
  slug: 'afkora',
  foundIn: [' , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.'],
}

describe('Search teaser component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<SearchTeaser data={result} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<SearchTeaser data={result} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct label', () => {
    const tree = renderer.create(<SearchTeaser data={result} />)
    const { root } = tree

    expect(root.findByType('a').children).toEqual([result.headword])
  })
})
