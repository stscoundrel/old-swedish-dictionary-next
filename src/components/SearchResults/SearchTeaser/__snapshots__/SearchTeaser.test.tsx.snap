// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search teaser component Matches snapshot 1`] = `
<div
  className="result"
>
  <a
    className="link"
    href="/word/afkora"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    afköra
  </a>
  <ul>
    <li
      className="foundIn"
      dangerouslySetInnerHTML={
        {
          "__html": " , vräka. hafwi wald han <mark>afköra</mark>  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.",
        }
      }
    />
  </ul>
</div>
`;
